content/enterprise_influxdb/v1.9/flux/optimize-queries.md

```diff
@@ -0,0 +1,182 @@
+Increase performance by avoiding short window durations.
+More windows require more compute power to evaluate which window each row should be assigned to.
+Reasonable window durations depend on the total time range queried.
+
+## Use "heavy" functions sparingly
+The following functions use more memory or CPU than others.
+Consider their necessity in your data processing before using them:
+
+- [map()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/map/)
+- [reduce()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/aggregates/reduce/)
+- [join()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/join/)
+- [union()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/union/)
+- [pivot()](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/pivot/)
+
+{{% note %}}
+We're continually optimizing Flux and this list may not represent its current state.
+{{% /note %}}
+
+## Use set() instead of map() when possible
+[`set()`](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/set/),
+[`experimental.set()`](/influxdb/v2.0/reference/flux/stdlib/experimental/set/),
+and [`map`](/influxdb/v2.0/reference/flux/stdlib/built-in/transformations/map/)
+can each set columns value in data, however **set** functions have performance
+advantages over `map()`.
+
+Use the following guidelines to determine which to use:
+
+- If setting a column value to a predefined, static value, use `set()` or `experimental.set()`.
+- If dynamically setting a column value using **existing row data**, use `map()`.
+
+#### Set a column value to a static value
+The following queries are functionally the same, but using `set()` is more performant than using `map()`.
+
+```js
+data
+  |> map(fn: (r) => ({ r with foo: "bar" }))
+
+// Recommended
+data
+  |> set(key: "foo", value: "bar")
+```
+
+#### Dynamically set a column value using existing row data
+```js
+data
+  |> map(fn: (r) => ({ r with foo: r.bar }))
+```
+
+## Balance time range and data precision
+To ensure queries are performant, balance the time range and the precision of your data.
+For example, if you query data stored every second and request six months worth of data,
+results would include ≈15.5 million points per series.  Depending on the number of series returned after `filter()`([cardinality](/influxdb/v2.0/reference/glossary/#series-cardinality)), this can quickly become many billions of points.
+Flux must store these points in memory to generate a response. Use [pushdowns](#pushdown-functions-and-function-combinations) to optimize how many points are stored in memory.
+
+To query data over large periods of time, create a task to [downsample data](/influxdb/v2.0/process-data/common-tasks/downsample-data/), and then query the downsampled data instead.
+
+## Measure query performance with Flux profilers
+Use the [Flux Profiler package](/influxdb/v2.0/reference/flux/stdlib/profiler/)
```

Comment:
not sure we have a link for this in /enterprise_influxdb/v1.9/ ??

Reply:
The flux stdlib docs live under 2.0, so this looks fine to me...